src/backend/distributed/commands/statistics.c

```diff
@@ -506,6 +518,65 @@ GetExplicitStatisticsSchemaIdList(Oid relationId)
 }
 
 
+/*
+ * GetAlterIndexStatisticsCommands returns the list of ALTER INDEX .. ALTER COLUMN ..
+ * SET STATISTICS commands, based on non default targets of the index with given id.
+ * Note that this function only looks for expression indexes, since this commands is
+ * valid for only expression indexes.
+ */
+static List *
+GetAlterIndexStatisticsCommands(Oid indexOid)
+{
+	List *alterIndexStatisticsCommandList = NIL;
+	int exprCount = 0;
+	while (true)
+	{
+		StringInfoData attName;
+		initStringInfo(&attName);
+
+		/* statistics can be set for an index only if it's an expressional index */
+		appendStringInfoString(&attName, "expr");
+
+		if (exprCount > 0)
+		{
+			appendStringInfo(&attName, "%d", exprCount);
+		}
```

Comment:
I think we can move this block -that builds attname string- into a function and the block that starts with `if (targetAttr->attstattarget != -1)` into a another function to simplify what this loop does

Reply:
could also define `-1` in a constant like `#define DEFAULT_STATISTIC_TARGET -1`